zealot/src/zng/types/record.ts

```diff
@@ -12,25 +12,29 @@ class UnknownColumnError extends Error {
 
 export type SerializedRecord = {
   type: zjson.Record
-  value: zjson.Value[]
+  value: zjson.Value[] | null
 }
 
 export class Record implements ZngClass<Type[] | null> {
-  constructor(readonly type: zjson.Column[], readonly value: zjson.Value[]) {}
+  constructor(
+    readonly type: zjson.Column[],
+    readonly value: zjson.Value[] | null
+  ) {}
 
   static deserialize({type, value}: SerializedRecord): Record {
     return new Record(type.of, value)
   }
 
   isSet() {
```

Comment:
```suggestion
  isSet(): this is {value: zjson.Value[]} {
```

You can fix the build errors with this type guard.